controllers/cloudflare_api.go

```diff
@@ -153,27 +102,20 @@ func (c *CloudflareAPI) DeleteCloudflareTunnel() error {
 		return err
 	}
 
-	req, _ := http.NewRequest("DELETE", fmt.Sprintf("%saccounts/%s/tunnels/%s", CLOUDFLARE_ENDPOINT, c.ValidAccountId, url.QueryEscape(c.ValidTunnelId)), nil)
-	if err := c.addAuthHeader(req, true); err != nil {
-		return err
-	}
+	ctx := context.Background()
```

Comment:
We do have a context from the Reconcile method, should we use the same?

Edit: It would need to pass through 3 different functions. I will make a separate PR for that actually.